### ISSUE TITLE
Compute BIC automatically from IBAN

### ISSUE THREAD

**Issue by mschulz-asphericon** [OCA/community-data-files#129]
regarding "base_bank_from_iban"

**Is your feature request related to a problem?**

Because the IBAN is sufficient to fetch the corresponding BIC and Country, it would be a great improvement to do this automatically.

Another problem is, that the module did not quite work for me, following the same steps as in the test instruction and the test file. For the exact same data it worked, but as soon as I have changed to German banks and partners it did not work correctly.

**Describe the solution you'd like**

On a partner's bank account list: when entering an IBAN it should check for available banks, if none is available it should suggest/ask if a new bank should be created with the information available from this API: 
https://openiban.com/validate/DE89370400440532013000?getBIC=true

**Describe alternatives you've considered**

**Additional context**

When creating SEPA mandates sometimes BIC codes are necessary, but these are not always available in the database and it is time consuming to check for the BIC codes, while this can be automated.

Looking forward to your response, best regards
Michael

**Comment by bosd** [comment]
1.
Using it on dutch IBAN'S and it doesn't work for me either.

The module stores the `code` in a separate field. While the code is already present in the IBAN.
Therefore better to change code to a computed field.
Add some logic to extract the bank code from the iban for the matching.

@oca/maintainers can you place a bug label on this issue?

2.
Your proposed solution is to validate/lookup with an external api. 
Looks cool, but should be a separate module.

**Comment by bosd** [comment]
Please reopen

**Comment by pedrobaeza** [comment]
Do you have any implementation?

**Comment by bosd** [comment]
> Do you have any implementation?

No, i don't have an emplementation.
Couple of days ago i stumbled upon this cool lib.


https://github.com/mdomke/schwifty

**Comment by pedrobaeza** [comment]
That library seems to do the same of the `base_bank_from_iban` module.

**Comment by bosd** [comment]
Yes, but the module is broken??
I don't remember correctly because it was some time ago.
I tested it manually against a bunch of different ibans from different countries.
After that ignored it, because it was not usable in my opinion. Similar to @mschulz-asphericon 

Ths does'nt mean the module does not work at all.. maybe against a very tiny subset of ibans.
(Which I did not find)

The earlier mentioned module can be an answer of importing bank accounts.
There was an discussion here, but i cannot find it again. 

About finding a good (open/free) data set for bank lists (with bic).
At the time of the discussion I could not find any good reliable source. even not at paid one. 
The problem is, that there is no official data set available. with good bank / bis lists.
Some data sets circulating on the internet/github is outright wrong.
(Even the commercial resellers of the free dataset) 


Wise.com was the best one according to my research.. But not open, and only available to its users..


This is the first project I have encounterd which actually produces a open and free dataset. Combining it from the official sources.

Right now our bank lists are handled in each localization repo.
Example: [l10n_nl_bank](https://github.com/OCA/l10n-netherlands/tree/14.0/l10n_nl_bank) to which I contributed a while ago. By manually converting the official file to odoo importable xml.
Problem is, that it only covers dutch bank list..
And that well.. as an integrator we do not go trough each and every repo to install the bank module.
We could benefit from a centralized community dataset of a bank list..
Have it updated automatically...

That lib, could satisfy both use cases
